Remove tensorFromBlob() from Type

li-roy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #18780 Remove tensorWithAllocator() from Type
* **#18779 Remove tensorFromBlob() from Type**

Removes tensorFromBlob() and puts equivalent functionality in at::from_blob(). 

Differential Revision: [D14739335](https://our.internmc.facebook.com/intern/diff/D14739335)

Commit:
Remove tensorFromBlob() from Type

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

ezyang commented on aten/src/ATen/TensorUtils.cpp:
```diff
@@ -235,4 +235,27 @@ bool geometry_is_contiguous(IntArrayRef sizes, IntArrayRef strides) {
+    stride *= sizes[i-1];
+  }
+  return strides;
+}
```
This is duplicate with this method in TensorImpl.h

```
  inline void update_to_contiguous_strides(size_t old_dim) {
    strides_.resize(sizes_.size(), 0);
    if (dim() > 0) {
      int last_idx = dim() - 1;
      strides_[last_idx] = 1;
      for (auto i = last_idx - 1; i >= 0; --i) {
        strides_[i] = strides_[i + 1] * std::max<int64_t>(sizes_[i + 1], 1);
      }
    }
    is_contiguous_ = true;
  }
```

ezyang replied:
I see the code existed from before. I'll accept the movement as long a we hide these methods in a private namespace (e.g., impl)

ezyang reviewed (approved):
LGTM modulo some naming comments.

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head

Commit:
Update on "Remove tensorFromBlob() from Type"

Remove tensorFromBlob() from Type

gh-metadata: pytorch pytorch 18779 gh/li-roy/2/head
